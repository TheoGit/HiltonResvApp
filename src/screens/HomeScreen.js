/**
 * @flow
 */

import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Button} from '../components/Button';

export default class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
  };

  onCreateReservationPress = () => {
    return this.props.navigation.navigate('Create');
  };

  onViewAllReservationsPress = () => {
    return this.props.navigation.navigate('ViewAll');
  };

  onGetByIdReservationsPress = () => {
    return this.props.navigation.navigate('ById');
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Theo's Reservation App</Text>
              <Button onPress={this.onCreateReservationPress}>Create</Button>
              <Button onPress={this.onViewAllReservationsPress}>View All</Button>
              <Button onPress={this.onGetByIdReservationsPress}>Get by Id</Button>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF', 
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
