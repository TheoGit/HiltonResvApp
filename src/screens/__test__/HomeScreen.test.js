import 'react-native';
import React from 'react';
import HomeScreen from '../HomeScreen';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import { shallow } from 'enzyme';

describe("HomeScreen", () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<HomeScreen navigation={navigation} />);

    it('HomeScreen renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("CreateReservation", () => {
        const spy = jest.spyOn(wrapper.instance(), "onCreateReservationPress");
        wrapper.instance().onCreateReservationPress();
        expect(spy).toHaveBeenCalled();
    });

    it("ViewAllReservations", () => {
        const spy = jest.spyOn(wrapper.instance(), "onViewAllReservationsPress");
        wrapper.instance().onViewAllReservationsPress();
        expect(spy).toHaveBeenCalled();
    });

    it("GetById", () => {
        const spy = jest.spyOn(wrapper.instance(), "onGetByIdReservationsPress");
        wrapper.instance().onGetByIdReservationsPress();
        expect(spy).toHaveBeenCalled();
    });
});