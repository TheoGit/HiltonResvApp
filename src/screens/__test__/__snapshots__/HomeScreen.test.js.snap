// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen HomeScreen renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Theo's Reservation App
        </Text>,
        <Button
          onPress={[Function]}
        >
          Create
        </Button>,
        <Button
          onPress={[Function]}
        >
          View All
        </Button>,
        <Button
          onPress={[Function]}
        >
          Get by Id
        </Button>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Theo's Reservation App",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Theo's Reservation App",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Create",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "Create",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "View All",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "View All",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Get by Id",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "Get by Id",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Theo's Reservation App
          </Text>,
          <Button
            onPress={[Function]}
          >
            Create
          </Button>,
          <Button
            onPress={[Function]}
          >
            View All
          </Button>,
          <Button
            onPress={[Function]}
          >
            Get by Id
          </Button>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Theo's Reservation App",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Theo's Reservation App",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Create",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Create",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "View All",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "View All",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Get by Id",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Get by Id",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
